src/AutoRest.CSharp/Common/Utilities/CachedDictionary.cs

```diff
@@ -25,11 +25,13 @@ public CachedDictionary(Func<Dictionary<K, V>> populateProc)
             _populateProc = populateProc;
         }
 
+        public bool IsPopulated => _values != null;
```

Comment:
Why do you need this?

Reply:
to allow us to partially build the parameters to get the path segments for a method before we fully initialize everything